### PR TITLE
Userpages - dump JSON data for each user

### PR DESCRIPTION
**Pull request by MariagraziaAlastra**
This change makes it possible to see the raw JSON data for each user, using the path /username/json
(eg '/jbarrett/json')

### CODE REVIEW

File: lib/DDH/UserPage/Generate.pm [hunk 1]
```diff
@@ -105,6 +110,19 @@ sub generate {
             }
         );
         close $fh;
+
+        # JSON dump of user's data
+        my $json_build_dir = $self->json_dir( $contributor );
+        open my $jfh, '>:encoding(UTF-8)', "$json_build_dir/index.html" or die();
+        print $jfh $self->xslate->render(
```
Comment (jbarrett) [inline lib/DDH/UserPage/Generate.pm]: Wondering if we could just dump index.json alongside index.html without a template? i.e. just the raw JSON as we get it?

Comment (MariagraziaAlastra) [inline lib/DDH/UserPage/Generate.pm]: @jbarrett Done, but for some reason it doesn't get nicely formatted like when I was using a template :(

File: lib/DDH/UserPage/Generate.pm [hunk 2]
```diff
@@ -105,6 +110,12 @@ sub generate {
             }
         );
         close $fh;
+
+        # JSON dump of user's data
+        my $json_build_dir = $self->json_dir( $contributor );
+        open my $jfh, '>:encoding(UTF-8)', "$json_build_dir/index.json" or die();
```
Comment (jbarrett) [inline lib/DDH/UserPage/Generate.pm]: Reckon it would be OK to just use `$build_dir`

Comment (jbarrett) [inline lib/DDH/UserPage/Generate.pm]: Reckon it would be OK to just use `$build_dir`?

Comment (MariagraziaAlastra) [inline lib/DDH/UserPage/Generate.pm]: Yeah, seems like it is, thanks!

**Comment by jbarrett** [comment]
:+1: Thanks, @MariagraziaAlastra!

**Pull request merged**